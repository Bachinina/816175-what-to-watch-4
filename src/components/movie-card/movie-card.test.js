import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";

const movie = {
  id: 1,
  bgColor: `#ef05fe`,
  bgImage: `img/bohemian-rhapsody-bg.jpg`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  director: `Wes Andreson`,
  genre: `Comedy`,
  isFavorite: false,
  name: `Bohemian Rhapsody`,
  posterImage: `img/bohemian-rhapsody-poster.jpg`,
  previewImage: `img/bohemian-rhapsody.jpg`,
  previewVideoLink: `https://some-link`,
  rating: 6.8,
  released: 1976,
  runTime: 74,
  scoresCount: 90,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  videoLink: `https://some-link`
};


it(`Should render MovieCard`, () => {
  const tree = renderer.create(
      <MovieCard
        movie={movie}
        onMovieCardMouseOver={() => {}}
        onMovieCardClick={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
