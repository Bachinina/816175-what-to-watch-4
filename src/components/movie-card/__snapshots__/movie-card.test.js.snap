// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MovieCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
    onClick={[Function]}
  >
    <img
      alt="Bohemian Rhapsody"
      height="175"
      src="img/bohemian-rhapsody.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
    >
      Bohemian Rhapsody
    </a>
  </h3>
</article>
`;
